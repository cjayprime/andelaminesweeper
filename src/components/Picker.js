import React from 'react';
import {TouchableOpacity, Text, Image} from 'react-native';

import Modal from './Modal';

import {checked} from '../assets/images';
import * as Style from '../assets/styles';

const Picker = (props) => {
  const {onChange, difficulty, open} = props;
  const options = ['Easy', 'Medium', 'Hard'];

  return (
    <Modal onClose={open}>
      {options.map((text, i) => (
        <TouchableOpacity
          key={i}
          onPress={() => onChange(text)}
          style={Style.picker.item}>
          <Text style={Style.picker.text}>{text}</Text>
          {text === difficulty && (
            <Image source={checked} style={Style.picker.image} />
          )}
        </TouchableOpacity>
      ))}
    </Modal>
  );
};

export default Picker;
