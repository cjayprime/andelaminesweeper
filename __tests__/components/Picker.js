import React from 'react';
import Picker from '../../src/components/Picker';

import renderer from 'react-test-renderer';

jest.useFakeTimers();
describe('Picker Test', () => {
    test('Stability', () => {
        const tree = renderer.create(
            <Picker
                open={() => {}}
                difficulty={'Easy'}
                onChange={() => {}}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});