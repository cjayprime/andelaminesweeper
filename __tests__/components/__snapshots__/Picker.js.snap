// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker Test Stability 1`] = `
Array [
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#000",
        "elevation": 100,
        "height": "100%",
        "left": 0,
        "opacity": 0.5,
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": 100000000000,
      }
    }
  />,
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderRadius": 10,
        "elevation": 100,
        "left": "20%",
        "padding": 5,
        "position": "absolute",
        "top": "30%",
        "width": "60%",
        "zIndex": 50000000000000000,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 15,
          }
        }
      >
        Easy
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/checked.png",
          }
        }
        style={
          Object {
            "height": 15,
            "width": 15,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 15,
          }
        }
      >
        Medium
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 15,
          }
        }
      >
        Hard
      </Text>
    </View>
  </View>,
]
`;
